src/jpeg/encoder.rs

```diff
@@ -405,14 +401,14 @@ impl<'a, W: Write> JpegEncoder<'a, W> {
             200 - scale * 2
         };
 
-        let mut tables = Vec::new();
-        let scale_value = |&v: &u8| {
-            let value = (u32::from(v) * scale + 50) / 100;
+        let mut tables = vec![STD_LUMA_QTABLE.clone(), STD_CHROMA_QTABLE.clone()];
+        let scale_value = |v: &mut u8| {
+            let value = (u32::from(*v) * scale + 50) / 100;
 
-            clamp(value, 1, u32::from(u8::max_value())) as u8
+            *v = clamp(value, 1, u32::from(u8::max_value())) as u8;
         };
-        tables.extend(STD_LUMA_QTABLE.iter().map(&scale_value));
-        tables.extend(STD_CHROMA_QTABLE.iter().map(&scale_value));
+        tables[0].iter_mut().for_each(scale_value);
+        tables[1].iter_mut().for_each(scale_value);
```

Comment:
This block of code could probably be cleaner with a bit of refactoring. Not a huge deal, but might be worth doing if it is being changed anyway

Reply:
What kind of reworking do you mean? Extracting the `scale_value` as a function?

Reply:
Mostly just removing duplication/making the code more concise. There's lots of possible variations, but one option would be something like:
```rust
let tables = vec![STD_LUMA_QTABLE.clone(), STD_CHROMA_QTABLE.clone()].into_iter().map(|t|t.iter_mut().for_each(|v|{
        *v = clamp((u32::from(*v) * scale + 50) / 100, 1, u32::from(u8::max_value())) as u8;
})).collect();

Reply:
Ahh, collecting the tables in one step instead of having it mutable and changing its contents. Sounds good

Reply:
I couldn't get it to be just one line without adding cloning, though I did remove the manual tables[0] / tables[1] scaling.  It's not particularly more concise, since maintaining readability also matters.